src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,286 @@
+          customHandler(id);
+        }
+        setSelected(id);
+      };
+
+      /**  Check props. */
+      const validateProps = (listItemsData, selectedItem) => {
+        if (!listItemsData) {
+          console.warn(
+            "NYPL Reservoir Menu: The `listItemsData` prop is required."
+          );
+        } else if (
+          selectedItem &&
+          !listItemsData.map((item) => item.id).includes(selectedItem)
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: The `selectedItem` prop does not match any of the menu items."
+          );
+        } else if (
+          new Set(listItemsData.map((item) => item.id)).size !==
+          listItemsData.length
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: The `id` values for the list items are not all unique."
+          );
+        } else if (
+          listItemsData.some(
+            (item) =>
+              !item.type ||
+              item.type === "" ||
+              !(
+                item.type === "group" ||
+                item.type === "action" ||
+                item.type === "divider"
+              )
+          )
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: A `type` value is required for each list item."
+          );
+        } else if (listItemsData.some((item) => !item.id || item.id === "")) {
+          console.warn(
+            "NYPL Reservoir Menu: An `id` value is required for each list item."
+          );
+        } else if (
+          listItemsData.some(
+            (item) =>
+              (item.type === "action" || item.type === "group") &&
+              (!item.label || item.label.trim() === "")
+          )
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: A `label` value is required for all list actions and groups."
+          );
+        } else if (
+          listItemsData.some((item) => item.type === "action" && !item.onClick)
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: An `onClick` function is required for all actions."
+          );
+        } else if (
+          listItemsData.some(
+            (item) =>
+              item.type === "group" &&
+              (!item.children || item.children.length === 0)
+          )
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: A `children` array is required for all list groups."
+          );
+        }
+        return true;
+      };
+      validateProps(listItemsData, selectedItem);
+
+      /** Helper function that renders either an Image or an Icon for a menu item. */
+      const renderMedia = (media: Media | null) => {
+        if (media) {
+          if (media.type === "icon") {
+            return <Icon name={media.name} size="medium" />;
+          } else {
+            return (
+              <Image
+                src={media.src}
+                alt={media.alt}
+                size="xxxsmall"
+                sx={{ borderRadius: "24px" }}
```

Comment:
This is fine but another option is setting `aspectRatio` to "square" and `imageType` to "circle"